Lib/urllib/parse.py

```diff
@@ -456,6 +456,11 @@ def urlsplit(url, scheme='', allow_fragments=True):
     """
 
     url, scheme, _coerce_result = _coerce_args(url, scheme)
+
+    for b in _UNSAFE_URL_BYTES_TO_REMOVE:
+        url = url.replace(b, "")
+        scheme = scheme.replace(b, "")
```

Comment:
According to the [typeshed](https://github.com/python/typeshed/blob/master/stdlib/urllib/parse.pyi#L126), the `scheme` argument to `urlparse` is `Optional`. However, with this change, a value of `None` will result in an error like:
```
AttributeError: 'NoneType' object has no attribute 'replace'
```
I'm not sure whether the typeshed is wrong or if this line should be guarded with `if scheme is not None`, but we should definitely fix one or the other and perhaps improve the documentation on this option.

Thanks to @branchvincent for doing the digging and reporting this issue to me.